[PyTorch] Use infer_size_dimvector in ExpandUtils

swolchok opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #55351 [PyTorch] Fix const correctness for resize native functions
* #55685 [PyTorch] Avoid double indirection in MaybeOwned's borrowed state
* #55684 [PyTorch] Change MaybeOwned tests to use intrusive_ptr and Tensor
* #55553 [PyTorch] Mark borrowed case as C10_LIKELY in MaybeOwned
* #55419 [PyTorch] Allow copy operations on MaybeOwned
* #55316 [PyTorch] Add & use inferExpandGeometry_dimvector
* #55247 [PyTorch][easy] Use sizes()[x] instead of size(x) in addr
* #55246 [PyTorch] Update expand_size API to match expand_inplace
* #55245 [PyTorch] Update expand_outplace API to match expand_inplace
* #55244 [PyTorch] Add MaybeOwned::operator*() &&
* **#55180 [PyTorch] Use infer_size_dimvector in ExpandUtils**
* #55065 [PyTorch] New expand_inplace API with MaybeOwned<Tensor> and no unary tuples

Even if we're expanding a Tensor's dimensions, DimVector's size is still a good guess at the rank of a Tensor in general. None of these sites actually seem to need a std::vector.

Differential Revision: [D27520127](https://our.internmc.facebook.com/intern/diff/D27520127/)

swolchok committed:
[PyTorch] Use infer_size_dimvector in ExpandUtils

Even if we're expanding a Tensor's dimensions, DimVector's size is still a good guess at the rank of a Tensor in general. None of these sites actually seem to need a std::vector.

Differential Revision: [D27520127](https://our.internmc.facebook.com/intern/diff/D27520127/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Use infer_size_dimvector in ExpandUtils"


Even if we're expanding a Tensor's dimensions, DimVector's size is still a good guess at the rank of a Tensor in general. None of these sites actually seem to need a std::vector.

Differential Revision: [D27520127](https://our.internmc.facebook.com/intern/diff/D27520127/)

[ghstack-poisoned]

swolchok committed:
rebase, fix typo in D27522811 on "[PyTorch] Use infer_size_dimvector in ExpandUtils"


Even if we're expanding a Tensor's dimensions, DimVector's size is still a good guess at the rank of a Tensor in general. None of these sites actually seem to need a std::vector.

Differential Revision: [D27520127](https://our.internmc.facebook.com/intern/diff/D27520127/)

[ghstack-poisoned]

ezyang commented on aten/src/ATen/ExpandUtils.h:
```diff
@@ -159,15 +159,15 @@ inline std::tuple<Tensor> expand_size(const Tensor &to_expand, IntArrayRef sizes
     if (!to_expand[i].defined()) {
       continue;
     } else if (first) {
       sizes = to_expand[i].sizes().vec();
```
the vec() call here looks sus

swolchok committed:
no more vec() call on "[PyTorch] Use infer_size_dimvector in ExpandUtils"


Even if we're expanding a Tensor's dimensions, DimVector's size is still a good guess at the rank of a Tensor in general. None of these sites actually seem to need a std::vector.

Differential Revision: [D27520127](https://our.internmc.facebook.com/intern/diff/D27520127/)

[ghstack-poisoned]

swolchok committed:
see if OSS doctest failures about non-ASCII character go away on "[PyTorch] Use infer_size_dimvector in ExpandUtils"


Even if we're expanding a Tensor's dimensions, DimVector's size is still a good guess at the rank of a Tensor in general. None of these sites actually seem to need a std::vector.

Differential Revision: [D27520127](https://our.internmc.facebook.com/intern/diff/D27520127/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Use infer_size_dimvector in ExpandUtils"


Even if we're expanding a Tensor's dimensions, DimVector's size is still a good guess at the rank of a Tensor in general. None of these sites actually seem to need a std::vector.

Differential Revision: [D27520127](https://our.internmc.facebook.com/intern/diff/D27520127/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Use infer_size_dimvector in ExpandUtils"


Even if we're expanding a Tensor's dimensions, DimVector's size is still a good guess at the rank of a Tensor in general. None of these sites actually seem to need a std::vector.

Differential Revision: [D27520127](https://our.internmc.facebook.com/intern/diff/D27520127/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Use infer_size_dimvector in ExpandUtils"


Even if we're expanding a Tensor's dimensions, DimVector's size is still a good guess at the rank of a Tensor in general. None of these sites actually seem to need a std::vector.

Differential Revision: [D27520127](https://our.internmc.facebook.com/intern/diff/D27520127/)

[ghstack-poisoned]
